generated/proto/policypb/policy.proto

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+option go_package = "github.com/m3db/m3metrics/generated/proto/policypb";
+
+package policypb;
+
+import "github.com/m3db/m3metrics/generated/proto/aggregationpb/aggregation.proto";
+
+message Resolution {
+  int64 window_size = 1;
+  int64 precision = 2;
+}
+
+message Retention {
+  int64 period = 1;
+}
+
+message StoragePolicy {
+  Resolution resolution = 1;
```

Comment:
nullable for this field?

Reply:
Not sure if I can since these are persisted in KV and I don't know if they are backwared compatible if I make this `nullable=false`.